packages/nuxt/src/components/scan.ts

```diff
@@ -137,3 +121,38 @@ export async function scanComponents (dirs: ComponentsDir[], srcDir: string): Pr
 
   return components
 }
+
+export function resolveComponentName (fileName: string, prefixParts: string[]) {
+  /**
+   * Array of fileName parts splitted by case, / or -
+   *
+   * @example third-component -> ['third', 'component']
+   * @example AwesomeComponent -> ['Awesome', 'Component']
+   */
+  const fileNameParts = splitByCase(fileName)
+
+  // e.g Item/Holder/Item/ItemHolderItem.vue -> ItemHolderItem
+  if (fileNameParts.join('').toLowerCase() === prefixParts.join('').toLowerCase()) {
+    return pascalCase(fileNameParts)
+  }
+
+  // e.g Item/Item/Item.vue -> Item
+  if ([...new Set(prefixParts)].length === 1 &&
+    (fileNameParts[0] || '').toLowerCase() === (prefixParts[0] || '').toLowerCase()) {
+    return pascalCase(fileNameParts)
+  }
```

Comment:
I'm worried these two sections are too 'specific' to the test cases I provided rather than reproducible as an algorithm.

I think:
* let's revert back to your original proposed change without this kind of 'special-case' handling
* update the test cases to accurately describe the previous behaviour with the exception of the bug reported initially (so we don't end up _breaking_ anyone's expected component resolution)